### PR TITLE
Added test to visualize lambda support

### LINKED ISSUE #66: Allow lambdas to be used everywhere
right now we require method references or lambdas with type information for most of the capture/call methods

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #66

### CODE REVIEW

File: src/test/java/org/zalando/riptide/LambdaTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import com.google.common.reflect.TypeToken;
+import org.junit.Test;
+import org.springframework.http.ResponseEntity;
+
+import static org.zalando.riptide.Conditions.anyStatus;
+import static org.zalando.riptide.Selectors.status;
+
+/**
+ * Tests whether all callbacks/functions can be supplied as lambdas. If this class compiles everything is already fine.
```
Comment (tkrop) [inline src/test/java/org/zalando/riptide/LambdaTest.java]: Funny. Compile time testing ... didn't even need to annotate with \@Test.

But, do we really need to test this explicitly?

Comment (lukasniemeier-zalando) [inline src/test/java/org/zalando/riptide/LambdaTest.java]: Users would be better off with a comprehensive Riptide example dispatch tree including some lambdas IMHO. I see value in this test class in a sense that future refactorings have to comply with it :wink:

Comment (whiskeysierra) [inline src/test/java/org/zalando/riptide/LambdaTest.java]: > Users would be better off with a comprehensive Riptide example dispatch tree including some lambdas IMHO

Improving the readme for 2.0.0 is still an open issue, see #76 and #67

Comment (whiskeysierra) [inline src/test/java/org/zalando/riptide/LambdaTest.java]: > I see value in this test class in a sense that future refactorings have to comply with it

Exactly, that was the idea, to have a regression test

**Comment by whiskeysierra** [comment]
Anyway, nobody said something about my question. Is the required type hint for the overloaded versions of capture/call ok?

**Comment by tkrop** [comment]
I can't see any difference in the behavior. Could you explain what I have missed?

**Comment by whiskeysierra** [comment]
> I can't see any difference in the behavior. Could you explain what I have missed?

The question was whether this would be ok:

```java
on(200).call(Success.class, (Success sucess) -> {..});
```

Because this doesn't work (due to method overloading):
```java
on(200).call(Success.class, sucess -> {..});
```

**Comment by tkrop** [comment]
Yes, but this does not work in the previous version either.

**Comment by tkrop** [comment]
:+1:

**Comment by whiskeysierra** [comment]
> Yes, but this does not work in the previous version either.

I know, I justed want to a) bring this to make this a conscious decision and b) to have a regression test that we don't loose the ability to supply lambdas in a future refactoring by accident.

**Comment by whiskeysierra** [comment]
Why is zappr not eating the approval?

**Comment by tkrop** [comment]
:+1:

**Pull request merged**